### PR TITLE
libsoup2.4-dev in 'rosdep/base.yml'

### PR DESCRIPTION
**Pull request by fbe555**
Please add the following dependency to the rosdep database.

## Package name:

libsoup2.4-dev

## Package Upstream Source:

https://gitlab.gnome.org/GNOME/libsoup

## Purpose of using this:

Needed as a system dependency for the [ros-gst-bridge](https://github.com/BrettRD/ros-gst-bridge) package 

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bookworm/libsoup2.4-dev
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/mantic/libsoup2.4-dev
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/libsoup/libsoup-devel/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/libsoup/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/net-libs/libsoup
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/libsoup@2#default
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://pkgs.alpinelinux.org/package/edge/community/x86_64/libsoup-dev
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=23.11&show=libsoup&from=0&size=50&sort=relevance&type=packages&query=libsoup
- openSUSE: https://software.opensuse.org/package/
  - https://software.opensuse.org/package/libsoup-2_4-1
- rhel: https://rhel.pkgs.org/
  - NOT AVAILABLE

### CODE REVIEW

**Review by audrow — DISMISSED** [review]

File: rosdep/base.yaml [hunk 1]
```diff
@@ -5753,6 +5753,16 @@ libsoundio-dev:
   macports: [libsoundio]
   nixos: [libsoundio]
   ubuntu: [libsoundio-dev]
+libsoup2.4-dev:
```
Comment (clalancette) [inline rosdep/base.yaml:5756]: I know that our typical policy is to make the key match the Ubuntu package name, but in this case I don't think it makes sense.  In particular, `libsoup-devel` on Fedora, for instance, is already libsoup 2.74.  So I'm going to suggest we call this key `libsoup-dev` instead.

**Review by clalancette — DISMISSED** [review]

File: rosdep/base.yaml (hunk 1 above)
Comment (fbe555) [reply rosdep/base.yaml]: Sounds resonable. I have renamed it.

Comment (sloretz) [reply rosdep/base.yaml]: @clalancette a downside of this is we now have a versionless key `libsoup-dev` and have to choose if or when  we'll upgrade it to the [libsoup-3.0-dev package](https://packages.debian.org/search?keywords=libsoup-3.0-dev).

**Review by sloretz — APPROVED** [review]
Changes look good to me, though I lean more towards making the rosdep key name match the debian package name for consistency.

File: rosdep/base.yaml [hunk 2]
```diff
@@ -5753,6 +5753,16 @@ libsoundio-dev:
   macports: [libsoundio]
   nixos: [libsoundio]
   ubuntu: [libsoundio-dev]
+libsoup-dev:
```
Comment (sloretz) [inline rosdep/base.yaml:5756]: @fbe555 -  thank you for your patience, would you be willing to change this name to `libsoup2-dev`?

Per offline discussion with @clalancette & @cottsay it seems like the best compromise we could come up with for handling multiple OS's packaging different versions of 2.x and 3.x of libsoup.

**Review by sloretz — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml (hunk 2 above)
Comment (fbe555) [reply rosdep/base.yaml]: Sure - that does make better sense. 
If you wanna reward my patience, pop by my other PR #40264. I Think it's ready to merge 😉

**Comment by fbe555** [comment]
Renamed it to libsoup2-dev now.

**Review by sloretz — APPROVED** [review]

**Review by cottsay — APPROVED** [review]

**Pull request merged**